(rewind pipe)add proper scroll

shaik-zeeshan opened the pull request:
added scroll based on scroll intensity

related issue: #1183
/claim #1183

https://github.com/user-attachments/assets/4bc309ec-8cd9-4c46-9b8a-1322005efbd6

louis030195 commented:
`when user arrive on the timeline it would be at current time and scrolling would make it go back in time before`

this imply you probably have to change this 

https://github.com/mediar-ai/screenpipe/blob/main/screenpipe-server/src/video_cache.rs

shaik-zeeshan commented:
thanks for pointing it out.

currently we are fetching data of 7 day from current time. should i keep this or change it to one day?

shaik-zeeshan commented:
i have changed data fetching to one day from current time 

if you want to change it , let me know

https://github.com/user-attachments/assets/4364609a-981a-48ea-8af4-e585aa7caba8

louis030195 commented:
nice for the order change

why does changing day started fetching again?

ideally we figure out how to load faster etc (read issue description)

shaik-zeeshan commented:
> i have changed data fetching to one day from current time

currently we are sending image data with frame data, if we remove it , we can get data faster

louis030195 commented:
i mean idk

this is not addressing the issue

shaik-zeeshan commented:
based on the issue , I have made the change .
to implement this I have changed ui , if you do not like it or have another design , let me know 

https://github.com/user-attachments/assets/cf8e40b5-8bd0-4db3-a58d-a8b0a6dd3a72

louis030195 commented:
just tested locally, this is really cool actually, great work!

can you invert the order though?

basically it should start from the right and when you scroll down it should go to the left (eg. "revisiting the past")

also if you can somehow show the icons back it would be nice, making it easier to see which app we were on from a high level glance

also the top level thing to move days is inverted to the bottom timeline (eg clicking left will go right)

btw did not understand these blue buttons at first:

<img width="86" alt="image" src="https://github.com/user-attachments/assets/8477f354-493e-46cd-a9cc-6718d1230b10" />

audio icon would make more sense IMHO!




<img width="1512" alt="Screenshot 2025-01-30 at 9 06 00 AM" src="https://github.com/user-attachments/assets/9ef3d4eb-118b-4e5d-93d2-b5c356a9519f" />

can you improve the UI when server is down?

louis030195 commented:
also would it be possible to detect if the user has enabled the `enable-frame-cache` in recording settings somehow and if not tell it to do so or do it for them? it's necessary to run the rewind thing

shaik-zeeshan commented:
> just tested locally, this is really cool actually, great work!

thanks

> can you invert the order though?
> 
> basically it should start from the right and when you scroll down it should go to the left (eg. "revisiting the past")
> 
> also if you can somehow show the icons back it would be nice, making it easier to see which app we were on from a high level glance
> 
> also the top level thing to move days is inverted to the bottom timeline (eg clicking left will go right)
> 
> btw did not understand these blue buttons at first:

inverted the direction and added app-icons , changed the indicator to more simple and understanding

https://github.com/user-attachments/assets/9f93eabb-7317-4c96-8f94-e996fe6a7e31





> can you improve the UI when server is down?
error is shown clearly
![Screenshot 2025-01-31 at 8 23 37 PM](https://github.com/user-attachments/assets/f93eeb6a-06aa-466c-aa35-3476762381bb)



> also would it be possible to detect if the user has enabled the `enable-frame-cache` in recording settings somehow and if not tell it to do so or do it for them? it's necessary to run the rewind thing

if enable-frame-cache is disabled this is shown
![Screenshot 2025-01-31 at 8 24 15 PM](https://github.com/user-attachments/assets/45e3a5f4-de68-4784-8491-4551eb363b6f)

louis030195 commented:
cool

can you solve conflicts and invert the scroll and i'll merge?

(eg scrolling down should go left, not up)

shaik-zeeshan commented:
> cool
> 
> can you solve conflicts and invert the scroll and i'll merge?
> 
> (eg scrolling down should go left, not up)

done

shaik-zeeshan commented:
wait don't merge there are some conflict ,  I am solving them

shaik-zeeshan commented:
done solved merge conflicts

louis030195 commented on screenpipe-app-tauri/src-tauri/src/server.rs:
```diff
@@ -113,31 +115,59 @@ async fn settings_stream(
-                    }
-                }
+
+    #[cfg(not(target_os = "windows"))]
```
issue in merge here

shaik-zeeshan replied:
fixed

louis030195 commented:
/approve

amazing!

tribhuwan-kumar commented:
@louis030195 
this pr has broke the ci for windows realease!
https://github.com/mediar-ai/screenpipe/actions/runs/13084067522/job/36512811695
![image](https://github.com/user-attachments/assets/af1676ef-c2d8-4a6f-b6b2-a280f2edae06)

shaik-zeeshan commented:
sorry guys , I couldn't test it on windows since I don't have windows 

doesn't the windows-icons library have get_icon_image_by_path method?
![Screenshot 2025-02-01 at 7 24 42 PM](https://github.com/user-attachments/assets/fc9c539b-e4bd-4ab8-bb94-348bc1adf7d9)

shaik-zeeshan commented:
@tribhuwan-kumar could you change that to get_icon_by_path ? check , if it works?

tribhuwan-kumar commented:
> @tribhuwan-kumar could you change that to get_icon_by_path ? check , if it works?

not going to work like that!!

louis030195 commented:
scroll does not work :( 

@shaik-zeeshan
